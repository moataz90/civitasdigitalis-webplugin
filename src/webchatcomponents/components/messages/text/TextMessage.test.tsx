import * as React from 'react';
import { TextMessage, Section } from './TextMessage';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

// The following test cases are based on jest snapshot testing!
// For more information on how snapshot tests are working, see
// https://jest-bot.github.io/jest/docs/snapshot-testing.html

test('render own text message is not last', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth={200}
			fontFamily="Helvetica"
			isOwnMessage={true}
			isLastMessage={false}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('render own text message is last', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth={200}
			fontFamily="Helvetica"
			isOwnMessage={true}
			isLastMessage={true}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('render bot text message is not last', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth={200}
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={false}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('render bot text message is last', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth={200}
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={true}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('max-width percentage', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={true}
			isLastMessage={false}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('own message and section is end', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={true}
			isLastMessage={false}
			section={Section.end}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('own message and section is single', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={true}
			isLastMessage={false}
			section={Section.end}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('bot message and section is end', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={false}
			section={Section.end}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('bot message and section is single', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={false}
			section={Section.single}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('bot message and section is start', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={false}
			section={Section.start}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('bot message and section is middle', () => {
	const domTree = renderer.create(
		<TextMessage
			text="Hello"
			fontColor="black"
			backgroundColor="white"
			maxWidth="50%"
			fontFamily="Helvetica"
			isOwnMessage={false}
			isLastMessage={false}
			section={Section.middle}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});