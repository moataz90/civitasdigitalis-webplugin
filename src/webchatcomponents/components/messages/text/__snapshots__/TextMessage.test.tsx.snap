// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bot message and section is end 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 3px 10px 6px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 0px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`bot message and section is middle 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 3px 10px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`bot message and section is single 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 8px 10px 6px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 0px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`bot message and section is start 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 8px 10px 3px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`max-width percentage 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`own message and section is end 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 3px 10px 6px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`own message and section is single 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  padding: 3px 10px 6px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 50%;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`render bot text message is last 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 200px;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`render bot text message is not last 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 200px;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": undefined,
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`render own text message is last 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 200px;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`render own text message is not last 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  max-width: 200px;
  font-size: 16px;
  line-height: normal;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 11px 18px 11px 18px;
  font-family: Helvetica;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  word-wrap: break-word;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;
